### PR TITLE
13.0 mig hr attendance report theoretical time

### PR DESCRIPTION
**Pull request by pegonzalezspesol**
Added oca_dependency.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
If I'm not wrong, since v12, the dependency to hr_holidays_public shouldn't be mandatory (it plugs in resource standard methods).

**Comment by pegonzalezspesol** [comment]
@angelmoya module installable, tests fixed and functionality reviewed. 
CI checks failed because of the dependency, how to deal with this @pedrobaeza ?

**Comment by jarroyomorales** [comment]
I'm afraid there is nothing we can do for now, just wait until `hr_holidays_public` gets migrated to 13.0

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py [hunk 1]
```diff
@@ -206,12 +212,14 @@ def read_group(
             orderby=orderby,
             lazy=lazy,
         )
-        if "theoretical_hours" not in fields:
+        if not any("theoretical_hours:sum" in field for field in fields):
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py]: Why not simply:
```python
if "theoretical_hours:sum" not in fields:
```
?

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml [hunk 2]
```diff
@@ -44,7 +44,7 @@
     <record id="hr_attendance_theoretical_action" model="ir.actions.act_window">
         <field name="name">Theoretical vs Attended Time Analysis</field>
         <field name="res_model">hr.attendance.theoretical.time.report</field>
-        <field name="view_type">form</field>
+
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml]: Don't add an empty line

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py [hunk 3]
```diff
@@ -141,13 +141,13 @@ def test_theoretical_hours(self):
         # 1946-12-23
         self.assertEqual(self.attendances[0].theoretical_hours, 8)
         self.assertEqual(self.attendances[1].theoretical_hours, 8)
-        # 1946-12-24
+        # # 1946-12-24
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: Why this extra comment symbol?

File: hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py [hunk 4]
```diff
@@ -229,7 +234,7 @@ def test_change_hr_holidays_public(self):
         # 1946-12-23
         self.assertEqual(self.attendances[0].theoretical_hours, 0)
         self.assertEqual(self.attendances[8].theoretical_hours, 0)
-        # 1946-12-25
+        # # 1946-12-25
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/tests/test_hr_attendance_report_theoretical_time.py]: Why extra comment symbol?

File: hr_attendance_report_theoretical_time/models/resource_calendar.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+    _description = "Resource Working Time"
+
+    def _work_intervals(self, start_dt, end_dt, resource=None, domain=None):
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/models/resource_calendar.py]: Why is this needed?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: hr_attendance_report_theoretical_time/README.rst [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fhr-lightgray.png?logo=github
+    :target: https://github.com/OCA/hr/tree/13.0/hr_attendance_report_theoretical_time
+    :alt: OCA/hr
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_attendance_report_theoretical_time
```
Comment (ernestotejeda) [inline hr_attendance_report_theoretical_time/README.rst]: There is still something referring to branch 12.0 in this line: .......hr-12-0/hr-12-0-hr_attendance_report_theoretical_time
Did you generate the readme using maintainer-tools module?

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py [hunk 7]
```diff
@@ -171,12 +170,19 @@ def _theoretical_hours(self, employee, date):
         """
         if not employee.resource_id.calendar_id:
             return 0
-        tz = employee.resource_id.calendar_id.tz
-        return employee.with_context(
-            exclude_public_holidays=True, employee_id=employee.id
-        ).get_work_days_data(
```
Comment (ernestotejeda) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py]: Why not change only get_work_days_data to _get_work_days_data

**Comment by rafaelbn** [comment]
Hello @pegonzalezspesol ! :-)

We are in a hurry with this migration. Please if you cannot attend it told us in order to continue it!

Thank you!

**Comment by pegonzalezspesol** [comment]
> Hello @pegonzalezspesol ! :-)
> 
> We are in a hurry with this migration. Please if you cannot attend it told us in order to continue it!
> 
> Thank you!

Hi @rafaelbn I currently can attend it. Otherwise I will let you know.

File: hr_attendance_report_theoretical_time/models/resource_calendar.py (hunk 5 above)
Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/models/resource_calendar.py]: if function _attendance_intervals does not receive a resource (in this case the employee) then a search on object "resource.calendar.attendance" with domain (resource_in in [False]) will be triggered with no result found. I'm opened to other solutions.

File: hr_attendance_report_theoretical_time/README.rst (hunk 6 above)
Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/README.rst]: Hi 
image

As far as I understand it I don't have to re-generate this file as it is automatically generated. @pedrobaeza am I wrong?

File: hr_attendance_report_theoretical_time/models/resource_calendar.py (hunk 5 above)
Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/models/resource_calendar.py]: as @ernestotejeda pointed in his comment using the private method _get_work_days_data fix this issue, so no need to rewrite the function _work_intervals

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py (hunk 7 above)
Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py]: thanks @ernestotejeda

File: hr_attendance_report_theoretical_time/README.rst (hunk 6 above)
Comment (chienandalu) [reply hr_attendance_report_theoretical_time/README.rst]: It reduces automatic commits noise though

File: hr_attendance_report_theoretical_time/wizards/recompute_theoretical_attendance_views.xml [hunk 8]
```diff
@@ -4,7 +4,7 @@
     <record id="act_wizard_recompute_theoretical_attendance" model="ir.actions.act_window">
         <field name="name">Recompute Theoretical Attendances</field>
         <field name="res_model">recompute.theoretical.attendance</field>
-        <field name="view_type">form</field>
+
```
Comment (chienandalu) [inline hr_attendance_report_theoretical_time/wizards/recompute_theoretical_attendance_views.xml]: don't leave an empty line, just remove the one you want to

**Review by chienandalu — COMMENTED** [review]
Looks good, although runbot isn't available for tests, so I couldn't make a functional one :confused:

File: hr_attendance_report_theoretical_time/wizards/recompute_theoretical_attendance_views.xml (hunk 8 above)
Comment (pegonzalezs) [reply hr_attendance_report_theoretical_time/wizards/recompute_theoretical_attendance_views.xml]: > don't leave an empty line, just remove the one you want to

done

File: hr_attendance_report_theoretical_time/README.rst (hunk 6 above)
Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/README.rst]: README.rst file generated with "oca-gen-addon-readme" tool.

@pedrobaeza should this step of generating the README file be included in the [tasks todo in migration](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#tasks-to-do-in-the-migration)?

Comment (pegonzalezspesol) [reply hr_attendance_report_theoretical_time/README.rst]: > It reduces automatic commits noise though

@chienandalu How can I get automatically rid of these noise commits? As far as I know, this is a manual "squash" process described in the "tasks todo in migration" section

**Review by ernestotejeda — APPROVED** [review]
LGTM

**Comment by chienandalu** [comment]
I'd like to do a little a functional test before approving. @pedrobaeza how's runbot troubleshooting status right now?

**Review by chienandalu — APPROVED** [review]
Ok, great. I finally could test it :slightly_smiling_face: :+1: Thanks for your work!

**Review by pedrobaeza — APPROVED** [review]

File: hr_attendance_report_theoretical_time/README.rst (hunk 6 above)
Comment (pedrobaeza) [reply hr_attendance_report_theoretical_time/README.rst]: @pegonzalezspesol it's not included in the steps of the migration guide because it's not mandatory. Indeed it's auto generated on merge, but pre-generating it reduces as said commit noise... and comments :stuck_out_tongue_closed_eyes:

**Comment by pegonzalezspesol** [comment]
> Ok, great. I finally could test it   Thanks for your work!

You are welcome @chienandalu . Thanks for your revision! Unfortunatelly I could not see what kind of noise is removed as many oca-bot automatic translations are still there, I will continue digging into.
